charts/turing/templates/turing-deployment.yaml

```diff
@@ -9,7 +9,7 @@ spec:
   replicas: {{ .Values.deployment.replicaCount }}
   selector:
     matchLabels:
-      app: {{ template "turing.fullname" . }}
```

Comment:
For Turing, the `app` label is also using the fullname, which needs to be updated to match the new selector: https://github.com/caraml-dev/helm-charts/blob/main/charts/turing/templates/_helpers.tpl#L12

Reply:
Ah that's true, I didn't change that initially because I was afraid it might break something else unintentionally and because this label wouldn't actually affect the behaviour of the deployment's pods but after checking further I think we can change that also and make it consistent with the other CaraML deployments. Thanks for pointing this out!

Reply:
In fact, only the Merlin and Turing deployments have this `app` label. XP and MLP don't have them 🤔

Reply:
I see.. So we are currently only setting the `app` label for the pods, in the case of MLP and XP but in the case of Merlin and Turing, it is set for all resources. I think it's nice to be consistent (either keep the `app` label only for the pods or for all components), but that's probably a little out of the scope of this MR. For now, I think it's reasonable to make all the Turing components' `app` labels to be the name rather than fullname.

Reply:
Yeah, another difference XP has from the other pods is that it's reusing the `management-svc.labels` block in its depeloyment pods, whereas the other CaraML app deployments (Turing, Merlin, MLP) only have `app` and `release` by default. Maybe we should iron out these differences to make it consistent across all the apps at some point of time 🤔